lib/scan/whatcms/engine.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Author: Pradeep Jairamani , github.com/pradeepjairamani
+
+import socket
+import socks
+import time
+import json
+import threading
+import string
+import random
+import sys
+import struct
+import re
+import os
+from OpenSSL import crypto
+import ssl
+from core.alert import *
+from core.targets import target_type
+from core.targets import target_to_host
+from core.load_modules import load_file_path
+from lib.socks_resolver.engine import getaddrinfo
+from core._time import now
+from core.log import __log_into_file
+import requests
+
+
+def extra_requirements_dict():
+    return {
+        "whatcms_ports": [443],
+        "whatcms_api_key": ["test_api_key"],
```

Comment:
best to keep  this empty and check that it is empty. I am currently getting NO ERROR when trying to use this modules without an API Key or with a WRONG API Key - something is not right with error handling

Reply:
Yes correct!!! This is the test_api_key(which is invalid), same as empty api key. Now we can check for that. If api key is invalid than the Nettacker will through invalid api key message :)